tactics/class_tactics.ml

```diff
@@ -198,15 +219,15 @@ let auto_core_unif_flags st freeze = {
   restrict_conv_on_strict_subterms = false; (* ? *)
   modulo_betaiota = true;
   modulo_eta = !typeclasses_modulo_eta;
-}
+  }
 
-let auto_unif_flags freeze st =
-  let fl = auto_core_unif_flags st freeze in
+let auto_unif_flags ~mode freeze st =
+  let fl = auto_core_unif_flags ~mode st freeze in
   { core_unify_flags = fl;
     merge_unify_flags = fl;
     subterm_unify_flags = fl;
     allow_K_in_toplevel_higher_order_unification = false;
-    resolve_evars = false
+    resolve_evars = (match mode with EautoCompat _ -> true | _ -> false)
```

Comment:
What does this flag mean and should it be instead

    resolve_evars = (match mode with OnlyClasses -> false | _ -> true)

?